### PR TITLE
Add TCP sender with connection reusage

### PR DESCRIPTION
**Pull request by medzin**
Add TCP sender with connection reusage

### CODE REVIEW

File: xnet/tcp.go [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+		closeErr := s.connections[addr].Close()
+		if closeErr != nil {
+			log.WithError(closeErr).Warn("Unable to close TCP connection properly")
+		}
+		delete(s.connections, addr)
+	}
+	return n, err
+}
+
+// Release frees system socket used by sender.
```
Comment (janisz) [inline xnet/tcp.go]: socket`s`

Comment (medzin) [reply xnet/tcp.go]: @janisz  fixed :)

File: xnet/tcp.go [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+		}
+		s.connections[addr] = newConn
+		conn = newConn
+	}
+	n, err := conn.Write(payload)
+	if err != nil {
+		// let's be nice and at least try to close connection on our side
+		closeErr := s.connections[addr].Close()
+		if closeErr != nil {
+			log.WithError(closeErr).Warn("Unable to close TCP connection properly")
```
Comment (AlfredBroda) [inline xnet/tcp.go]: We will see this message before the actual error that might have caused it (above). Why not log the above message first too? Or return a MultiError.

Comment (medzin) [reply xnet/tcp.go]: @AlfredBroda fixed

**Pull request merged**